src/main/domain/vendorlist/VendorListHelper.js

```diff
@@ -0,0 +1,48 @@
+export class VendorListHelper {
+  haveAllValuesTo({object, valueToVerify}) {
+    if (Object.entries(object).length === 0) {
+      return valueToVerify === false
+    }
+    // eslint-disable-next-line no-unused-vars
+    for (const [_, value] of Object.entries(object)) {
+      if (valueToVerify !== value) {
+        return false
+      }
+    }
+    return true
+  }
+
+  setAllVendorsTo({vendorList, valueToSet}) {
+    const newVendor = {
+      consents: {},
+      legitimateInterests: {}
+    }
+    Object.entries(vendorList).forEach(([key, value]) => {
+      newVendor.consents[key] = valueToSet
+      newVendor.legitimateInterests[key] = valueToSet
+    })
+    return newVendor
+  }
+
+  mergeVendors({newVendorList, oldVendors}) {
+    const mergedVendorList = {
+      consents: {},
+      legitimateInterests: {}
+    }
+
+    Object.entries(newVendorList).forEach(([key, value]) => {
```

Comment:
Same as before with Object Keys. Also, why do you need to parseInt the key?

Reply:
Ok, using Object keys, and Ok removing parse int. done